staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -1446,14 +1446,29 @@ func (az *Cloud) reconcileLoadBalancer(clusterName string, service *v1.Service,
 				}
 				klog.V(10).Infof("EnsureBackendPoolDeleted(%s) for service %s: end", lbBackendPoolID, serviceName)
 
-				// Remove the LB.
-				klog.V(10).Infof("reconcileLoadBalancer: az.DeleteLB(%q): start", lbName)
-				err = az.DeleteLB(service, lbName)
+				existingLBs, err := az.ListLB(service)
 				if err != nil {
```

Comment:
Alternatively, is it possible to check for an error here that indicates "LB not found" and we can return nil, nil in that case?

Reply:
Yeah that would work as well. But I think a LIST here would also work.

Reply:
Checking an error from delete would remove the need for an additional API call though right?

Reply:
Not going to hold the cherry-pick on it, but could be worth revisting

Reply:
Delete is much easier to get throttled compared to GET requests

Reply:
Yeah, in fact the purpose of this change is to reduce the delete call.